app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Admin::DealsController < Admin::BaseController
+  before_action :set_deal, only: %i[show edit update destroy check_publishablity]
+  
+  def index
+    @deals = Deal.all_deals
+  end
+
+  def new
+    @deal = Deal.new
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def create
+    @deal = Deal.new(deal_params)
+    @deal.calculate_tax_on_deal
```

Comment:
this should be done via a before_validation callback

Reply:
done